core/t_set.go

```diff
@@ -0,0 +1,290 @@
+package core
+
+import (
+	"errors"
+	"sort"
+)
+
+func evalSADD(args []string) []byte {
+	if len(args) < 2 {
+		return Encode(errors.New("ERR wrong number of arguments for 'SADD' command"), false)
+	}
+	key := args[0]
+
+	// Get the set object from the store.
+	obj := Get(key)
+
+	var count int = 0
+	if obj == nil {
+		var exDurationMs int64 = -1
+		var keepttl bool = false
+		// If the object does not exist, create a new set object.
+		value := make(map[string]bool)
+
+		// Add the value to the set.
+		for i := 1; i < len(args); i++ {
+			value[args[i]] = true
+			count++
+		}
+
+		// Create a new object.
+		Put(key, NewObj(value, exDurationMs, ObjTypeSet, ObjEncodingHT), WithKeepTTL(keepttl))
+	} else {
+		// If the object exists, check if it is a set object.
+		if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+			return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+		}
+
+		// Get the set object.
+		set := obj.Value.(map[string]bool)
+
+		// Add the value to the set.
+		for i := 1; i < len(args); i++ {
+			if _, ok := set[args[i]]; !ok {
+				set[args[i]] = true
+				count++
+			}
+		}
+	}
+	return Encode(count, false)
+}
+
+func evalSMEMBERS(args []string) []byte {
+	if len(args) != 1 {
+		return Encode(errors.New("ERR wrong number of arguments for 'SMEMBERS' command"), false)
+	}
+	key := args[0]
+
+	// Get the set object from the store.
+	obj := Get(key)
+
+	if obj == nil {
+		return Encode([]string{}, false)
+	}
+
+	// If the object exists, check if it is a set object.
+	if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+		return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+	}
+
+	// Get the set object.
+	set := obj.Value.(map[string]bool)
+
+	// Get the members of the set.
+	var members []string
+	for k, flag := range set {
+		if flag {
+			members = append(members, k)
+		}
+	}
+
+	return Encode(members, false)
+}
+
+func evalSREM(args []string) []byte {
+	if len(args) < 2 {
+		return Encode(errors.New("ERR wrong number of arguments for 'SREM' command"), false)
+	}
+	key := args[0]
+
+	// Get the set object from the store.
+	obj := Get(key)
+
+	var count int = 0
+	if obj == nil {
+		return Encode(count, false)
+	}
+	if obj.TypeEncoding != ObjTypeSet|ObjEncodingHT {
+		return Encode(errors.New("WRONGTYPE Operation against a key holding the wrong kind of value"), false)
+	}
+
+	// Get the set object.
+	set := obj.Value.(map[string]bool)
+	for i := 1; i < len(args); i++ {
+		if set[args[i]] {
+			delete(set, args[i])
+			count++
+		}
+	}
```

Comment:
```suggestion
	for _, arg := range args[1:] {
	if set[arg] {
		delete(set, arg)
		count++
	}
}

```
same for all other such loops in this PR.